x-pack/plugins/ml/public/application/data_frame_analytics/common/use_results_view_config.ts

```diff
@@ -54,6 +64,29 @@ export const useResultsViewConfig = (jobId: string) => {
           analyticsConfigs.data_frame_analytics.length > 0
         ) {
           const jobConfigUpdate = analyticsConfigs.data_frame_analytics[0];
+          const analysisType = getAnalysisType(jobConfigUpdate.analysis);
+          // don't fetch the total feature importance if it's outlier_detection
+          if (
+            analysisType === ANALYSIS_CONFIG_TYPE.CLASSIFICATION ||
+            analysisType === ANALYSIS_CONFIG_TYPE.REGRESSION
+          ) {
+            try {
+              const inferenceModels = await inferenceApiService.getInferenceModel(`${jobId}*`, {
+                include: 'total_feature_importance',
+              });
+              const inferenceModel = inferenceModels.find(
+                (model) => model.metadata?.analytics_config?.id === jobId
+              );
+              if (
+                Array.isArray(inferenceModel?.metadata?.total_feature_importance) === true &&
```

Comment:
Nit - I don't think you really need the `=== true` after the `isArray` check as the `isArray` method is pretty clear.

Reply:
For some reasons TS needs the explicit check here to make the second part of the if statement valid. Not sure why but I will leave it for now if that's okay.